### PR TITLE
Hba improvements

### PR DESCRIPTION
**Pull request by pgiraud**

### CODE REVIEW

File: pgtoolkit/hba.py [hunk 1]
```diff
@@ -382,6 +382,8 @@ def remove(
         :param attrs: keyword/values pairs correspond to one or more
             HBARecord attributes (ie. user, conntype, etc...)
 
+        :returns ``True`` if records have changed.
```
Comment (dlax) [inline pgtoolkit/hba.py:388]: There should be a `:` after `returns` for Sphinx.

File: pgtoolkit/hba.py [hunk 2]
```diff
@@ -411,14 +414,16 @@ def remove(
...
+    def merge(self, other: "HBA") -> bool:
         """Add new records to HBAFile or replace them if they are matching
             (ie. same conntype, database, user and address)
 
         :param other: HBAFile to merge into the current one.
             Lines with matching conntype, database, user and database will be
             replaced by the new one. Otherwise they will be added at the end.
             Comments from the original hba are preserved.
+
+        :returns ``True`` if records have changed.
```
Comment (dlax) [inline pgtoolkit/hba.py:426]: Same here.

File: pgtoolkit/hba.py [hunk 3]
```diff
@@ -236,6 +236,9 @@ def __str__(self) -> str:
 
         return line
 
+    def __eq__(self, other: object) -> bool:
+        return str(self) == str(other)
+
```
Comment (dlax) [inline pgtoolkit/hba.py:241]: Should be in a dedicated commit, with some test. Or did you commit this by mistake?

Comment (pgiraud) [reply pgtoolkit/hba.py:241]: I'll add tests and put it in a separated commit.

File: pgtoolkit/hba.py [hunk 4]
```diff
@@ -455,13 +455,16 @@ def merge(self, other: "HBA") -> bool:
...
 
-    :param file: Either a line iterator such as a file-like object or a string
+    :param file: Either a line iterator such as a file-like object, a path or a string
         corresponding to the path to the file to open and parse.
     :rtype: :class:`HBA`.
     """
+    if isinstance(file, Path):
+        file = str(file)
+
     if isinstance(file, str):
```
Comment (dlax) [inline pgtoolkit/hba.py:468]: I think `isinstance(file, (str, Path)):` is sufficient as `open()` can handle a `Path` value, so no need to convert the `Path` into a string.

File: tests/test_hba.py [hunk 5]
```diff
@@ -201,6 +201,12 @@ def test_parse_file(mocker):
     pgpass.lines.append(HBAComment("# Something"))
     assert m.called
 
+    # Also works with path
+    pgpass = parse(tmp_path / "filename")
+    pgpass.lines.append(HBAComment("# Something"))
```
Comment (dlax) [inline tests/test_hba.py:206]: `m.reset_mock()` missing

**Review by dlax — CHANGES_REQUESTED** [review]

**Review by dlax — APPROVED** [review]
